pkg/compute/kube/client.go

```diff
@@ -141,6 +139,27 @@ func (c *client) GetServicesForServiceIdentity(svcIdentity identity.ServiceIdent
 	return meshServices
 }
 
+// GetServicesForProxy maps an Envoy instance to a number of Kubernetes services.
+func (c *client) GetServicesForProxy(p *envoy.Proxy) ([]service.MeshService, error) {
```

Comment:
Not something to fix in this PR, but we should try and avoid using this function as much as possible or finally fix #1684. A loop through k8s service in our cache is a bottleneck for large clusters